include/sys/vdev_disk.h

```diff
@@ -42,12 +42,17 @@
 
 #ifdef _KERNEL
 #include <sys/vdev.h>
+#include <sys/vdev_disk_os.h>
 
 typedef struct vdev_disk {
+	/* Common members */
 	ddi_devid_t		vd_devid;
 	char			*vd_minor;
```

Comment:
It looks like neither `vd_devid` or `vd_minor` are used at all on Linux or FreeBSD.  Moreover, the `vdev_disk_t` is only used on Linux, FreeBSD stashes a `g_consumer` struct in the `vd->vdev_tsd`.  Rather than try and make this common what if we moved the contents of `vdev_disk.h` in to `module/os/linux/zfs/vdev_disk.c` (and removed the unused members).  Then for macos you can do something similar and define whatever structures you need.

Reply:
OK, define the struct in the linux .c file? osx will make an /os/ header, since it is shared with the ldi_compat code.

Reply:
Yes, it looks like it's only ever used in the `.c` so that should be absolutely fine.  If it makes sense to put it in an /os/ header for osx that seems fine too.